services/src/pro/datasets/storage.rs

```diff
@@ -0,0 +1,197 @@
+use crate::datasets::storage::MetaDataDefinition;
+use crate::datasets::upload::UploadId;
+use crate::datasets::{AddDataset, DatasetIdAndName};
+use crate::error::Result;
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::DatasetId;
+use geoengine_datatypes::primitives::DateTime;
+use postgres_types::{FromSql, ToSql};
+use serde::{Deserialize, Serialize};
+use tokio_postgres::Transaction;
+use utoipa::ToSchema;
+
+#[derive(Deserialize, Serialize, Debug, Clone, ToSchema)]
+pub struct Expiration {
+    pub deletion_timestamp: Option<DateTime>,
+    pub delete_record: bool,
+    pub delete_data: bool,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct ChangeDatasetExpiration {
+    pub dataset_id: DatasetId,
+    pub expiration_change: ExpirationChange,
+}
+
+impl ChangeDatasetExpiration {
+    pub fn delete_meta(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: true,
+                delete_data: false,
+            }),
+        }
+    }
+
+    pub fn delete_fair(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: false,
+                delete_data: true,
+            }),
+        }
+    }
+
+    pub fn delete_full(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: true,
+                delete_data: true,
+            }),
+        }
+    }
+
+    pub fn delete_none(dataset_id: DatasetId) -> Self {
```

Comment:
when would this be used and not expire_none?

Reply:
When the deletion of meta data should happen immediately. But since this is no longer an option, this might be resolved.